src/ray/core_worker/actor_manager.cc

```diff
@@ -168,4 +185,21 @@ std::vector<ObjectID> ActorManager::GetActorHandleIDsFromHandles() {
   return actor_handle_ids;
 }
 
+ActorID ActorManager::GetCachedNamedActorID(const std::string &actor_name) {
+  {
+    absl::MutexLock lock(&cache_mutex_);
+    auto it = cached_actor_name_to_ids_.find(actor_name);
+    if (it != cached_actor_name_to_ids_.end()) {
+      absl::MutexLock lock(&mutex_);
+      auto handle_it = actor_handles_.find(it->second);
+      if (handle_it == actor_handles_.end()) {
+        cached_actor_name_to_ids_.erase(actor_name);
```

Comment:
What's the case that the handle is in `cached_actor_name_to_ids_` but not in `actor_handles_`?

If we have no such case, maybe it's better to `CHECK()` to let the bug easy to figure.

Reply:
Alright. Such codes are used for sanity. CHECK() is better.